src/taxpasta/infrastructure/application/kaiju_profile.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2022, Moritz E. Beber, Maxime Borry, Jianhong Ou, Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a description of the kaiju profile format."""
+
+
+from typing import Optional
+from unicodedata import category
+
+import numpy as np
+import pandas as pd
+import pandera as pa
+from pandera.typing import Series
+
+
+class KaijuProfile(pa.SchemaModel):
+    """Define the expected kaiju profile format."""
+
+    file: Series[str] = pa.Field()
+    percent: Series[float] = pa.Field(ge=0.0, le=100.0)
+    reads: Series[int] = pa.Field(ge=0)
+    taxonomy_id: Series[pd.CategoricalDtype] = pa.Field()
```

Comment:
@Midnighter There are some NAs in the test data and I have been trying to handle them.

I tried with `taxonomy_id: Series[float] = pa.Field(nullable=True)` as well. Would you suggest that I should handle the NAs before defining the SchemaModel?

Reply:
There are NAs in the taxonomy ID column or in the reads/percent column?

Reply:
NAs in the taxonomy ID column.

`barcode41_se-barcode41-kaiju.tsv	59.486836	16739	NA	unclassified`

Reply:
I see, for unclassified. Kraken uses ID -1 for that. There is no way to make Kaiju use the empty string instead of NA, is there?

Reply:
Not to my knowledge.

I was thinking if there is a way of replacing the NAs with 0 in the SchemaModel. I see that the `nullable=True` can be accepted as an argument and that it should not be a problem but that is not the case here.

Reply:
Do you mean to use `fillna("-1")` in `taxonomy_id: Series[pd.CategoricalDtype] = pa.Field()` ?